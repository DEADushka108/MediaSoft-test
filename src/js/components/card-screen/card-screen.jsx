import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header';
import Footer from '../footer/footer';
import { getActiveRace, getCardCount, getFilteredList, getShipsList } from '../../store/ships/selectors';
import { connect } from 'react-redux';
import { ActionCreator as Ships } from '../../store/ships/ships';


const CardScreen = (props) => {
  const {ship, onAddCartClick} = props;
  const {id, name, image, description, price, isInCart, rating} = ship;

  return <Fragment>
    <Header/>

    <main className="page-main">
      <div className="page-main__wrapper">
        <h1 className="visually-hidden">Ship card</h1>
        <section className="ship-card ship-card--full">
          <div className="ship-card__hero">
            <div className="ship-card__bg">
              <img className="ship-card__img" src={image} alt={name} />
            </div>
          </div>
          <div className="ship-card__wrapper">
            <div className="ship-card__rating">
              <p className="ship-card__text">Rating: {rating}</p>
            </div>
            <div className="ship-card__price">
              <p className="ship-card__text">Price: {price}$</p>
            </div>
            <div className="ship-card__description">
              <p className="ship-card__text">Description: {description}</p>
            </div>
          </div>
          <button className="ship-card__button" type="button" onClick={() => {
            onAddCartClick(ship);
          }}>
            {isInCart ? `In Cart` : `Add to cart`}
          </button>
        </section>
      </div>
    </main>

    <Footer/>

  </Fragment>
};

CardScreen.propTypes = {};

const mapDispatchToProps = (dispatch) => ({
  onAddCartClick(ship) {
    ship.isInCart = !ship.isInCart;
    dispatch(Ships.updateShipStatus(ship));
  }
});

export {CardScreen};
export default connect(null, mapDispatchToProps)(CardScreen);
