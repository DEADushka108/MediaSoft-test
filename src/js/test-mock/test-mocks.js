export * from './const/const';
export * from './ships/ships';
export * from './store/store';
