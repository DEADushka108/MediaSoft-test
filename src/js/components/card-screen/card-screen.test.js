import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {ship, noop, store} from '../../test-mock/test-mocks';
import {CardScreen} from './card-screen';

it(`Should CardScreen component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <CardScreen
            ship={ship}
            onAddCartClick={noop}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
