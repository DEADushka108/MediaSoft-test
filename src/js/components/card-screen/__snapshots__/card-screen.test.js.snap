// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CardScreen component render correctly 1`] = `
Array [
  <header
    className="page-header"
  >
    <div
      className="page-header__wrapper"
    >
      <a
        className="page-header__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link main-nav__link--active"
              href="/"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#home-page"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Shop
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/contact"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#social-group"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Contact
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/cart"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#shopping-cart"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Cart
              </span>
              <span
                className="main-nav__sub"
              >
                2
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <div
      className="page-main__wrapper"
    >
      <h1
        className="visually-hidden"
      >
        Ship card
      </h1>
      <section
        className="ship-card ship-card--full"
      >
        <div
          className="ship-card__hero"
        >
          <div
            className="ship-card__bg"
          >
            <img
              alt="Battlestar Galactica"
              className="ship-card__img"
              src="./img/bg/battlestar-galactica.png"
            />
          </div>
        </div>
        <div
          className="ship-card__wrapper"
        >
          <div
            className="ship-card__rating"
          >
            <p
              className="ship-card__text"
            >
              Rating: 
              6.1
            </p>
          </div>
          <div
            className="ship-card__price"
          >
            <p
              className="ship-card__text"
            >
              Price: 
              234122
              $
            </p>
          </div>
          <div
            className="ship-card__description"
          >
            <p
              className="ship-card__text"
            >
              Description: 
              Battlespaceship
            </p>
          </div>
        </div>
        <button
          className="ship-card__button"
          onClick={[Function]}
          type="button"
        >
          Add to cart
        </button>
      </section>
    </div>
  </main>,
  <footer
    className="page-footer"
  >
    <div
      className="page-footer__wrapper"
    >
      <a
        className="page-footer__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <div
        className="page-footer__copyright"
      >
        <p
          className="page-footer__text"
        >
          Â© 2020 Stars ships shop Ltd.
        </p>
      </div>
    </div>
  </footer>,
]
`;
